src/components/HTMLEngineProvider/HTMLRenderers/PreRenderer/index.js

```diff
@@ -1,68 +1,66 @@
-import React from 'react';
+import React, {useCallback, useEffect, useRef} from 'react';
 import _ from 'underscore';
+
+import ControlSelection from '../../../../libs/ControlSelection';
+import * as DeviceCapabilities from '../../../../libs/DeviceCapabilities';
 import withLocalize from '../../../withLocalize';
 import htmlRendererPropTypes from '../htmlRendererPropTypes';
 import BasePreRenderer from './BasePreRenderer';
-import * as DeviceCapabilities from '../../../../libs/DeviceCapabilities';
-import ControlSelection from '../../../../libs/ControlSelection';
 
-class PreRenderer extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.scrollNode = this.scrollNode.bind(this);
-        this.debouncedIsScrollingVertically = _.debounce(this.isScrollingVertically.bind(this), 100, true);
-    }
-
-    componentDidMount() {
-        if (!this.ref) {
-            return;
-        }
-        this.ref.getScrollableNode().addEventListener('wheel', this.scrollNode);
-    }
-
-    componentWillUnmount() {
-        this.ref.getScrollableNode().removeEventListener('wheel', this.scrollNode);
-    }
+function PreRenderer(props) {
+    const scrollViewRef = useRef();
 
     /**
      * Check if user is scrolling vertically based on deltaX and deltaY. We debounce this
      * method in the constructor to make sure it's called only for the first event.
      * @param {WheelEvent} event Wheel event
      * @returns {Boolean} true if user is scrolling vertically
      */
-    isScrollingVertically(event) {
+    function isScrollingVertically(event) {
```

Comment:
we use `const` arrow functions 

```
const isScrollingVertically = (event) => {}
```

Reply:
also this function must use `useCallback` because is needed for `debouncedIsScrollingVertically` if not that function will be recreated on every rerender